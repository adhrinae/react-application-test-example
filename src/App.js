// @ts-check
import React, { Component } from 'react';
import axios from 'axios';

export const doIncrement = prevState => ({
  counter: prevState.counter + 1,
});

export const doDecrement = prevState => ({
  counter: prevState.counter - 1,
});

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      counter: 0,
      asyncCounters: null,
    };
  }

  onIncrement = () => {
    this.setState(doIncrement);
  };

  onDecrement = () => {
    this.setState(doDecrement);
  };

  componentDidMount() {
    axios
      .get('https://swapi.co/api/films')
      .then(data => {
        this.setState({ asyncCounters: data });
      })
      .catch(error => console.error(error));
  }

  render() {
    const { counter } = this.state;

    return (
      <div>
        <h1>My Counter</h1>
        <Counter counter={counter} />

        <button type="button" onClick={this.onIncrement}>
          Increment
        </button>

        <button type="button" onClick={this.onDecrement}>
          Decrement
        </button>
      </div>
    );
  }
}

export const Counter = ({ counter }) => <p>{counter}</p>;

export default App;
