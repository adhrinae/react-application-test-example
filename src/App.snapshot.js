import React from 'react';
import renderer from 'react-test-renderer';

import App, { Counter } from './App';

describe('App Snapshot', () => {
  test('renders', () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Counter Snapshot', () => {
  test('renders', () => {
    const component = renderer.create(<Counter counter={1} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
