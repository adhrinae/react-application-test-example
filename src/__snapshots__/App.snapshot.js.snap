// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot renders 1`] = `
<div>
  <h1>
    My Counter
  </h1>
  <p>
    0
  </p>
  <button
    onClick={[Function]}
    type="button"
  >
    Increment
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Decrement
  </button>
</div>
`;

exports[`Counter Snapshot renders 1`] = `
<p>
  1
</p>
`;
